docopt does not compile with ghc-8.8.2

~~~~
[ 6 of 10] Compiling System.Console.Docopt.Public ( System/Console/Docopt/Public.hs, dist/build/System/Console/Docopt/Public.o )

System/Console/Docopt/Public.hs:195:17: error:
    • Could not deduce (MonadFail m) arising from a use of ‘fail’
      from the context: Monad m
        bound by the type signature for:
                   getFirstArg :: forall (m :: * -> *).
                                  Monad m =>
                                  Arguments -> Option -> m String
        at System/Console/Docopt/Public.hs:193:1-57
      Possible fix:
        add (MonadFail m) to the context of
          the inferred type of failure :: m a
          or the type signature for:
               getFirstArg :: forall (m :: * -> *).
                              Monad m =>
                              Arguments -> Option -> m String
    • In the expression: fail $ "no argument given: " ++ show opt
      In an equation for ‘failure’:
          failure = fail $ "no argument given: " ++ show opt
      In the expression:
        let failure = fail $ "no argument given: " ++ show opt
        in
          case opt `M.lookup` args of
            Nothing -> failure
            Just val
              -> case val of
                   MultiValue vs -> ...
                   Value v -> return v
                   _ -> failure
    |
195 |   let failure = fail $ "no argument given: " ++ show opt
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 7 of 10] Compiling System.Console.Docopt.UsageParse ( System/Console/Docopt/UsageParse.hs, dist/build/System/Console/Docopt/UsageParse.o )

System/Console/Docopt/UsageParse.hs:7:1: warning: [-Wunused-imports]
    The import of ‘dropWhile’ from module ‘Data.List’ is redundant
  |
7 | import           Data.List (nub, sortBy, maximumBy, dropWhile, dropWhileEnd)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 8 of 10] Compiling System.Console.Docopt.QQ ( System/Console/Docopt/QQ.hs, dist/build/System/Console/Docopt/QQ.o )

System/Console/Docopt/QQ.hs:59:25: error:
    • No instance for (MonadFail ((->) String))
        arising from a use of ‘fail’
    • In the first argument of ‘(.)’, namely ‘fail’
      In the expression: fail . (++ " context unsupported")
      In an equation for ‘unsupported’:
          unsupported = fail . (++ " context unsupported")
   |
59 |     where unsupported = fail . (++ " context unsupported")
   |                         ^^^^
~~~~

--- a/System/Console/Docopt/Public.hs
+++ b/System/Console/Docopt/Public.hs
@@ -190,7 +190,7 @@ isPresentM :: Monad m => Arguments -> Option -> m Bool
 isPresentM args o = return $ isPresent args o
 
 {-# DEPRECATED getFirstArg "Use 'getAllArgs' instead" #-}
-getFirstArg :: Monad m => Arguments -> Option -> m String
+getFirstArg :: MonadFail m => Arguments -> Option -> m String
 getFirstArg args opt =
   let failure = fail $ "no argument given: " ++ show opt
   in  case opt `M.lookup` args of

--- a/System/Console/Docopt/QQ.hs
+++ b/System/Console/Docopt/QQ.hs
@@ -56,7 +56,9 @@ docopt = QuasiQuoter { quoteExp  = docoptExp
                      , quotePat  = unsupported "Pattern"
                      , quoteType = unsupported "Type"
                      }
-    where unsupported = fail . (++ " context unsupported")
+    where unsupported :: String -> String -> Q a
+          unsupported qqType _ = do
+            fail $ (qqType ++ " context unsupported")
 
 -- | Same as 'docopt', but parses the given file instead of a literal
 -- string.

--- a/docopt.cabal
+++ b/docopt.cabal
@@ -43,7 +43,7 @@ library
                       System.Console.Docopt.OptParse
                       System.Console.Docopt.Public
 
-  build-depends:      base >= 4.0 && < 5.0,
+  build-depends:      base >= 4.9 && < 5.0,
                       parsec >= 3.1.0,
                       containers
 

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,7 +1,7 @@
 # For advanced use and comprehensive documentation of the format, please see:
 # http://docs.haskellstack.org/en/stable/yaml_configuration/
 
-resolver: lts-7.4
+resolver: lts-16.20
 
 packages:
   - '.'

